### DISCUSSION TITLE
Strange TextEditor behavior, insight requested

### DISCUSSION THREAD

**Discussion by djarb** [fltk-rs/fltk-rs#1483] [Q&A] [answered]
Does anybody have any ideas about what's happening here? ![weird](https://github.com/fltk-rs/fltk-rs/assets/150819/08eef4dc-888d-4477-8013-d7969e891d5d)

The screen shot shows the bottom left corner of a Text Editor widget. When the widget is first shown, the frame is drawn as is should be. 

Then, when the mouse moves into the widget or the focus is moved, the frame is replaced by heavy black bars, as seen in the screenshot.

Clicking in the text edit area restores the correct frame, but moving the mouse out and back in brings back the heavy black bars, as does moving the focus again.

So, the problem is pretty clearly related to focus, but what's not clear to me is what's actually going wrong. I've looked through the FLTK source code for the text editor widget and its ancestors without finding anything that could explain this.

**Comment by MoAlyousef** [comment]
I’ve seen a similar artifact when using the wayland backend on linux. 
I thought it might be a compisitor issue. Is this on linux. Are you building with use-wayland. 
Can you provide more info on your environment?

**Reply by djarb** [reply]
It is on Linux and I am using Wayland, but I've already tested on XWayland and plain old X11 and seen identical results.

The build does include the use-wayland feature. I just tested, and when rebuilt without use-wayland the effect disappears, even when running via XWayland. That's a good workaround to know about, but I would still like for it to work properly with use-wayland enabled if we can figure out how.

Also, thank you for taking the time to help.

**Reply by MoAlyousef** [reply]
Sorry I was a bit busy the last few days.
Can you provide the current distro you're using, its version along with the wayland compositor. 
I'll raise this on the fltk mailing list.

**Reply by djarb** [reply]
Not a problem, you don't owe me anything, and I appreciate what you do.

I'm using Arch Linux (synced daily) and the compositor is Hyprland from the Arch package extra/hyprland 0.33.1-2

I don't think the compositor is relevant, though, because the same behavior was exhibited even under X11 as long as the use-wayland feature (and thus, presumably, the code for the Wayland backend of FLTK) was enabled. My guess: there's some wonky drawing logic that gets enabled with that backend.

**Reply by MoAlyousef** [reply, marked as answer]
I think it was from the code to support opacity in fltk-rs. Can you try with master?

**Reply by djarb** [reply]
Looks like you're right about that. I recompiled with the git master version, and the weird effect has gone away.

**Reply by djarb** [reply]
By the way, if libdecor is installed in the system, it appears to cause linker errors when the use-wayland feature is enabled. That's unrelated to the main topic of this thread, but I discovered it while testing the solution.

**Reply by MoAlyousef** [reply]
It seems FLTK has recently added a cmake option OPTION_USE_SYSTEM_LIBDECOR which is by default ON. This will cause linking issues.
I've changed it to OFF while configuring CMake. I'm not sure whether it's useful to add a feature to fltk-rs to toggle this option on and off.

**Reply by djarb** [reply]
Thank you again for helping me understand the problem, then fixing it, then fixing another problem too. You're great!
